server/lib/allowed-features.ts

```diff
@@ -38,22 +92,33 @@ export const hasOptedInForFeature = (collective, feature): boolean => {
 };
 
 /**
- * Check if the given feature is activated for collective.
+ * If a given feature is allowed for the collective type, check if it is activated for collective.
  */
-const hasFeature = (collective, feature: FEATURE): boolean => {
+export const hasFeature = (collective, feature: FEATURE): boolean => {
   if (!collective) {
     return false;
   }
 
-  // Check type
-  if (!isFeatureAllowedForCollectiveType(collective.type, feature)) {
-    return false;
+  // Check opt-out flags
+  if (feature in OPT_OUT_FEATURE_FLAGS) {
+    return !hasOptedOutOfFeature(collective, feature);
   }
 
-  // Check opt-out flags
-  return !hasOptedOutOfFeature(collective, feature);
+  // Check opt-in flags
+  if (feature in OPT_IN_FEATURE_FLAGS) {
+    return hasOptedInForFeature(collective, feature);
+  }
+
+  // Check if collective feature is disabled
+  if (feature in OPTIONAL_COLLECTIVE_FEATURES) {
+    const collectivePageSections = get(collective, 'settings.collectivePage.sections');
+    if (collectivePageSections) {
+      const featureSettings = collectivePageSections.find(s => s.section === OPTIONAL_COLLECTIVE_FEATURES[feature]);
+      return featureSettings.isEnabled;
+    }
+  }
```

Comment:
This is a good approach toward solving https://github.com/opencollective/opencollective/issues/3396, though as mentioned in my comment there I wonder how this will play with the `CONNECT` section. Let's discuss this point later.

Reply:
As discussed, have put it back to `settings.features.FEATURE` for now